### PR TITLE
Fix register writing on Windows

### PR DESCRIPTION
**Pull request by GustavoLCR**

### CODE REVIEW

**Comment by XVilka** [comment]
Please rebase, since I merged your AppVeyor fix

**Comment by GustavoLCR** [comment]
> Please rebase, since I merged your AppVeyor fix

@XVilka no need, Appveyor was already working for PRs, its failing because of master:
```
[XX] db\io\baddr bin.baddr and anal  317
$ r2 -escr.utf8=0 -escr.color=0 -escr.interactive=0 -N -Q -B 0xf00000 -i C:\Users\appveyor\AppData\Local\Temp\1\tmp-3884ktAQ0kEp69FJ ../bins/elf/analysis/pie
aaa
afl
+0x00f00450    1 49           entry0
 0x00f00482    1 4            fcn.00f00482
 0x00f00440    1 6            sym.imp.__libc_start_main
 0x00f005c5    1 4            main
 0x00f004a0    4 61           sym.deregister_tm_clones
 0x00f00490    1 4            sym.__x86.get_pc_thunk.bx
 0x00f004e0    4 71           sym.register_tm_clones
 0x00f00530    5 71           entry.fini0
 0x00f00420    1 6            sym.imp.__cxa_finalize
 0x00f00580    4 69   -> 60   entry.init0
 0x00f00640    1 2            sym.__libc_csu_fini
 0x00f00644    1 20           sym._fini
 0x00f005d0    4 97           sym.__libc_csu_init
 0x00f003ec    3 35           sym._init
 0x00f00430    1 6            loc.imp.__gmon_start
-0x00f00450    1 49           entry0
```

**Comment by XVilka** [comment]
This one was fixed.

File: libr/debug/p/native/windows/windows_debug.c [hunk 1]
```diff
@@ -455,6 +455,15 @@ int w32_reg_read(RDebug *dbg, int type, ut8 *buf, int size) {
 	return size;
 }
 
+static void __transfer_drx(RDebug *dbg, ut8 *buf) {
+	CONTEXT cur_ctx;
+	if (w32_reg_read (dbg, R_REG_TYPE_ALL, &cur_ctx, sizeof (CONTEXT))) {
+		CONTEXT *new_ctx = (CONTEXT *)buf;
+		memcpy (&cur_ctx.Dr0, &new_ctx->Dr0, offsetof (CONTEXT, Dr7) - offsetof (CONTEXT, Dr0) + sizeof (new_ctx->Dr7));
```
Comment (XVilka) [inline libr/debug/p/native/windows/windows_debug.c]: can you split this line? e.g. calculate the offset before memcpy?

**Pull request merged**